src/liballoc/boxed.rs

```diff
@@ -91,17 +91,16 @@ use str::from_boxed_utf8_unchecked;
 #[unstable(feature = "box_heap",
            reason = "may be renamed; uncertain about custom allocator design",
            issue = "27779")]
-pub const HEAP: ExchangeHeapSingleton = ExchangeHeapSingleton { _force_singleton: () };
+pub const HEAP: ExchangeHeapSingleton = ExchangeHeapSingleton { };
 
 /// This the singleton type used solely for `boxed::HEAP`.
 #[unstable(feature = "box_heap",
            reason = "may be renamed; uncertain about custom allocator design",
            issue = "27779")]
 #[allow(missing_debug_implementations)]
 #[derive(Copy, Clone)]
-pub struct ExchangeHeapSingleton {
-    _force_singleton: (),
-}
+#[non_exhaustive]
+pub struct ExchangeHeapSingleton { }
```

Comment:
No need to use the braced form for `non_exhaustive` structs, more usual `pub struct ExchangeHeapSingleton;` can be used.

Reply:
Done; I mostly did this because I wasn't sure if this had been implemented yet. Although in honesty, the goal should be to make sure this feature works by first testing it in the standard library.